support/@trustfractal/data-host/src/popup/services/Factory.ts

```diff
@@ -0,0 +1,106 @@
+import { ApiPromise, WsProvider } from '@polkadot/api';
+import environment from '@popup/Environment';
+import { DataHost } from '@services/DataHost';
+import { FractalAccountConnector } from '@services/FractalAccount';
+import { ProtocolService } from '@services/Protocol';
+import { ProtocolOptIn } from '@services/ProtocolOptIn';
+import { StorageService } from '@services/StorageService';
+import { AggregateMultiContext, MultiContext } from '@utils/MultiContext';
+import { ValueCache } from '@utils/ReactHooks';
+
+const types = {
+  FractalId: 'u64',
+  MerkleTree: 'Raw',
+};
+
+let storageService: StorageService;
```

Comment:
I'm not super happy with having this one factory file. I'd prefer a dependency injection framework, but I didn't want to do the effort of integrating that into the wallet. We can consider changing this.

Reply:
Do you have on mind a specific dependency injection framework, or should I research?

Reply:
I like Angular's, but I don't know how usable it is outside of Angular projects. Not worth holding up this PR though.